docs/specification/appendix.md

```diff
@@ -65,15 +65,43 @@ Schema `encoding` may only be omitted for self-describing message encodings such
 
 ### ros1msg
 
-- `name`: A valid [package resource name](http://wiki.ros.org/Names#Package_Resource_Names), e.g. `sensor_msgs/PointCloud2`
+- `name`: A valid [package resource name](http://wiki.ros.org/Names#Package_Resource_Names), e.g. `sensor_msgs/PointCloud2`. The package resource name does not include the file extension.
 - `encoding`: `ros1msg`
-- `data`: Concatenated ROS1 .msg files
+- `data`: Delimited concatenated ROS1 .msg files.
+
+#### ros1msg Data Format
+
+the `data` field contains the concatenated `.msg` file content that is sent in the ROS subscription connection header for this message type.
+
+The top-level message definition is present first, with no delimiter. All dependent `.msg` definitions are preceded by a two-line delimiter:
+
+- One line containing exactly 80 `=` characters
+- One line containing `MSG: <package resource name>` for that type.
+
+This format can be reproduced using [gendeps --cat](http://wiki.ros.org/roslib/gentools).
 
 ### ros2msg
 
-- `name`: A valid [package resource name](http://wiki.ros.org/Names#Package_Resource_Names), e.g. `sensor_msgs/msg/PointCloud2`
+- `name`: A valid [package resource name](http://wiki.ros.org/Names#Package_Resource_Names), e.g. `sensor_msgs/msg/PointCloud2`.
 - `encoding`: `ros2msg`
-- `data`: Concatenated ROS2 .msg files
+- `data`: Delimited concatenated ROS2 .msg files
+
+#### ros2msg Data Format
+
+The `.msg` definition is stored alongside its dependencies in the same format as [ros1msg](#ros1msg-data-format).
+
+### ros2idl
+
+- `name`: A valid [package resource name](http://wiki.ros.org/Names#Package_Resource_Names), e.g. `sensor_msgs/msg/PointCloud2`
+- `encoding`: `ros2idl`
+- `data`: ROS2 .idl files, concatenated with [`gendeps`-style message definition delimiting](#gendeps-style-message-definition-delimiting)
```

Comment:
Is `#gendeps-style-message-definition-delimiting` suppose to reference something? I don't think that exists in this document anymore?

Reply:
Out of date, let me remove.